ast/parser.go

```diff
@@ -849,6 +849,13 @@ func (p *Parser) parseLiteral() (expr *Expr) {
 					p.hint("`import future.keywords.every` for `every x in xs { ... }` expressions")
 				}
 			}
+			// If we can't find `every or some` still check if import future.keywords.in
+			if term, ok := expr.Terms.(*Term); ok && (!Var("every").Equal(term.Value) && !Var("some").Equal(term.Value)) {
+				imp := p.parseImport()
```

Comment:
Now I'm confused... How is `p.parseImport` what we'd do here?

If the parser encounters this rule,

```rego
p {
  "foo" in {"bar": "foo"}
}
```
and it can't make sense of it because `"foo" in {"bar": "foo"}` isn't parsed without `in` being a keyword... how is parsing this as an import the right thing? Imports within rule bodies are never valid... 🤔

Reply:
I think p.parseImport returns nil in the case of import error or no import, the purpose of this change is only to check whether import future.keywords or future.keywords.in

Reply:
I'm not sure that this is the correct approach... `p.parseImport` will never be non-nil, I think. And it would thus more often record the hint than it should... but I'm guessing a bit here, I haven't dug in yet to see if that's the case... 🔍

Reply:
Thanks for bearing with me.
I'll keep thinking about the relevant code. See if there is a better solution